File: telerik_ui_for_winforms_courseware_chm/Courseware/Grid/VB/RadGridView/DynamicLINQLibrary/RadGridViewLab10.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Linq.Dynamic
Imports System.Windows.Forms

Namespace _10_DynamicLINQLibrary
  Public Partial Class RadGridViewLab10
	  Inherits Form
	Public Sub New()
	  InitializeComponent()
	End Sub

	Private Sub btnSort_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnSort.Click
            Me.radGridView1.TableElement.BeginUpdate()
	  Dim queryable As IQueryable = New SalesOrderHeaderDataClassDataContext().SalesOrderHeaders.AsQueryable()
	  queryable = queryable.OrderBy(String.Format("{0} {1}", cbField.Text, cbSortType.Text))
	  radGridView1.DataSource = queryable.Take(Convert.ToInt32(numRecords.Value))
            Me.radGridView1.TableElement.EndUpdate(True)
	End Sub

  End Class
End Namespace

File: telerik_ui_for_winforms_courseware_chm/Courseware/Grid/CS/RadGridView/DynamicLINQLibrary/RadGridViewLab10.cs
using System;
using System.Linq;
using System.Linq.Dynamic;
using System.Windows.Forms;

namespace _10_DynamicLINQLibrary
{
    public partial class RadGridViewLab10 : Form
    {
        public RadGridViewLab10()
        {
            InitializeComponent();
        }

        private void btnSort_Click(object sender, EventArgs e)
        {
            this.radGridView1.TableElement.BeginUpdate();
            IQueryable queryable = new SalesOrderHeaderDataClassDataContext().SalesOrderHeaders.AsQueryable();
            queryable = queryable.OrderBy(String.Format("{0} {1}",
            cbField.Text, cbSortType.Text));
            radGridView1.DataSource =
            queryable.Take(Convert.ToInt32(numRecords.Value));
            this.radGridView1.TableElement.EndUpdate(true);
        }
    }
}
